File: backend/testcases/C/medium.c

// --- Helper function to check if two integers are equal (for tests) ---

bool intsEqual(int a, int b) {
    return a == b;
}

// --- Test main() ---

int main() {
    bool allTestsPassed = true;

    MinStack* minStack = minStackCreate();

    minStackPush(minStack, -2);
    minStackPush(minStack, 0);
    minStackPush(minStack, -3);

    if (!intsEqual(minStackGetMin(minStack), -3)) {
        allTestsPassed = false;
    }

    minStackPop(minStack);

    if (!intsEqual(minStackTop(minStack), 0)) {
        allTestsPassed = false;
    }

    if (!intsEqual(minStackGetMin(minStack), -2)) {
        allTestsPassed = false;
    }

    minStackFree(minStack);

    if (allTestsPassed) {
        printf("ALL_TESTS_PASSED\n");
    } else {
        printf("FAIL\n");
    }

    return 0;
}

File: backend/testcases/java/medium.java
public class Main {
    public static void main(String[] args) {
        try {
            MinStack MinStack = new MinStack();
            MinStack.push(-2);de
            MinStack.push(0);
            MinStack.push(-3);

            assert MinStack.getMin() == -3 : "Test 1 Failed";

            MinStack.pop();

            assert MinStack.top() == 0 : "Test 2 Failed";
            assert MinStack.getMin() == -2 : "Test 3 Failed";

            System.out.println("ALL_TESTS_PASSED");
        } catch (AssertionError e) {
            System.out.println("FAIL");
        }
    }
}